server/jetstream_cluster.go

```diff
@@ -1006,6 +1006,16 @@ func (cc *jetStreamCluster) isConsumerLeader(account, stream, consumer string) b
 	return false
 }
 
+func (cc *jetStreamCluster) nodeForConsumerProposals(sa *streamAssignment) RaftNode {
```

Comment:
Could you add a comment about the JS lock needing to be held? (I assume it's already held at all call sites, but just for documentation)